src/System.Runtime.Extensions/tests/System/IO/PathTestsBase.cs

```diff
@@ -205,6 +205,174 @@ public partial class PathTestsBase : RemoteExecutorTestBase
+            { @"C:\git\corefx\..\corefx\.\..\corefx", @"C:\git\corefx", @"C:\git\corefx" },
+            { @"C:\somedir\..", @"C:\git\corefx", @"C:\" },
+            { @"C:\", @"C:\git\corefx", @"C:\" },
+            { @"..\..\..\..", @"C:\git\corefx", @"C:\" },
+            { @"C:\\\", @"C:\git\corefx", @"C:\" },
+            { @"C:\..\..\", @"C:\git\corefx", @"C:\" },
+            { @"C:\..\git\..\.\", @"C:\git\corefx", @"C:\" },
+            { @"C:\git\corefx\..\..\..\", @"C:\git\corefx", @"C:\" },
+            { @"C:\.\corefx\", @"C:\git\corefx", @"C:\corefx\" },
+        };
+
+        public static TheoryData<string, string, string> GetFullPath_CommonUnRootedWindowsData => new TheoryData<string, string, string>
+        {
+            { "", @"C:\git\corefx", @"C:\git\corefx" },
+            { "..", @"C:\git\corefx", @"C:\git" },
+
+            // Current drive rooted
+            { @"\tmp\bar", @"C:\git\corefx", @"C:\tmp\bar" },
+            { @"\.\bar", @"C:\git\corefx", @"C:\bar" },
+            { @"\tmp\..", @"C:\git\corefx", @"C:\" }, 
+            { @"\tmp\bar\..", @"C:\git\corefx", @"C:\tmp" },
+            { @"\tmp\bar\..", @"C:\git\corefx", @"C:\tmp" },
+            { @"\", @"C:\git\corefx", @"C:\" },
+
+            // Specific drive rooted
+            { @"C:tmp\foo\..", @"C:\git\corefx", @"C:\git\corefx\tmp" },
+            { @"C:tmp\foo\.", @"C:\git\corefx", @"C:\git\corefx\tmp\foo" },
+            { @"C:tmp\foo\..", @"C:\git\corefx", @"C:\git\corefx\tmp" },
+            { @"C:tmp", @"C:\git\corefx", @"C:\git\corefx\tmp" },
+            { @"C:", @"C:\git\corefx", @"C:\git\corefx" },
+            { @"C", @"C:\git\corefx", @"C:\git\corefx\C" },
+
+            { @"Z:tmp\foo\..", @"C:\git\corefx", @"Z:\tmp" },
+            { @"Z:tmp\foo\.", @"C:\git\corefx", @"Z:\tmp\foo" },
+            { @"Z:tmp\foo\..", @"C:\git\corefx", @"Z:\tmp" },
+            { @"Z:tmp", @"C:\git\corefx", @"Z:\tmp" },
+            { @"Z:", @"C:\git\corefx", @"Z:\" },
+            { @"Z", @"C:\git\corefx", @"C:\git\corefx\Z" }
+        };
+
+        public static TheoryData<string, string, string> GetFullPath_Windows_UNC => new TheoryData<string, string, string>
+        {
+            { @"foo", @"", @"foo" },
+            { @"foo", @"server1", @"server1\foo" },
+            { @"\foo", @"server2", @"server2\foo" },
+            { @"foo", @"server3\", @"server3\foo" },
+            { @"..\foo", @"server4", @"server4\..\foo" },
+            { @".\foo", @"server5\share", @"server5\share\foo" },
+            { @"..\foo", @"server6\share", @"server6\share\foo" },
+            { @"\foo", @"a\b\\", @"a\b\foo" },            
+            { @"foo", @"LOCALHOST\share8\test.txt.~SS", @"LOCALHOST\share8\test.txt.~SS\foo" },
+            { @"foo", @"LOCALHOST\share9", @"LOCALHOST\share9\foo" },
+            { @"foo", @"LOCALHOST\shareA\dir", @"LOCALHOST\shareA\dir\foo" },
+            { @". \foo", @"LOCALHOST\shareB\", @"LOCALHOST\shareB\. \foo" },
```

Comment:
I don't see any tests that attempt to simply eat into the Server\Share.  E.g. "..", "..\..".